### ISSUE TITLE
t-less datasets pose ground truth

### ISSUE THREAD

**Issue by firstdeep** [zju3dv/clean-pvnet#5]
Thank you for your wonderful results and code sharing.

Currently, I want to proceed customer learning using tless dataset.
I saw the file t-less_01.yaml in your configs folder.

Therefore, I would like to ask you how you constructed the pose ground truth.
If you have sample ground truth for Tless 01, I would appreciate it if you could share it with me.

P.S I created a pose ground truth using the value of the gt.yml file given in the tless dataset. Is this the right way?

I'll wait for your reply.

Thank you.

**Comment by pengsida** [comment]
You are right. I have been performing experiments on the tless dataset.
I will upload an instruction for training the linemod and pretrained models.
A related code can be found at https://github.com/zju3dv/clean-pvnet/blob/master/lib/datasets/tless/handle_test_data.py#L212

**Comment by pengsida** [comment]
Training data: https://github.com/zju3dv/clean-pvnet/blob/master/lib/datasets/tless/tless_to_coco.py#L104

**Comment by firstdeep** [comment]
Thank you for your reply :)

Your research is very good.
I used the 23rd object of tless and achieved 2d projections metric 0.96 in 140th time of epoch.

I wonder if it is right to learn so quickly.
The performance is so amazing.

Thank you

**Comment by pengsida** [comment]
I am not sure.
I did not upload the evaluation code for the tless.
I just began to train the tless.

**Comment by firstdeep** [comment]
Oh, really?

I conducted the train and test using the "Training on the custom object" item you gave me. 

I'll be waiting for your tless instruction. 
Thank you.

**Comment by pengsida** [comment]
However, the tless has many scenes, and each scene has many instances of the same object.
I do not know what data you use for training and testing.
I will complete the training in two months.

**Comment by firstdeep** [comment]
I conducted the training using only 1296 RGB images of object 23 of Microsoft KINect v2.

The test used a tless test image.

I just want to find the posture of one object.

Of course, it depends on the orientation of the study, 
but is it OK?? to use the reading "Training on the Custom object" you wrote in the README to conduct the data train?

**Comment by pengsida** [comment]
A test image?
The custom dataset is used for the case that there is only an instance of the target object in the image.

**Comment by firstdeep** [comment]
I'll check the code a little bit more.

Thank you for your kind answer.

**Comment by firstdeep** [comment]
@pengsida 
> Training data: https://github.com/zju3dv/clean-pvnet/blob/master/lib/datasets/tless/tless_to_coco.py#L113

I wonder how to calculate "farthest.txt" in tless datasets

thank you

**Comment by pengsida** [comment]
I will upload the dataset and pretrained models tomorrow.

**Comment by pengsida** [comment]
I sampled the farthest points using https://github.com/zju3dv/clean-pvnet/blob/bbaf334fd58e134af350877fc19a9efa1db212b2/tools/handle_custom_dataset.py#L19

**Comment by firstdeep** [comment]
@pengsida 

Thank you very much.

I would also appreciate it if you could tell me how to train & test the tless datasets tomorrow.
I'm having a hard time at the test.

Thank you 

Have a good night.

-david

**Comment by pengsida** [comment]
I have uploaded the code, pretrained models and data about tless.

**Comment by firstdeep** [comment]
@pengsida 

Thank you so much for sharing the tless code.

I am going to use the kinect camera datasets. Therefore, I would like to ask about the format of the tless datasets and how it is generated.

thank you 

david

**Comment by pengsida** [comment]
I process the tless data using https://github.com/zju3dv/clean-pvnet/blob/master/run.py#L174
`handle_test_data` processes the test data.
`handle_ag_data` and `tless_to_coco` processes the training data for detection and pose estimation.
`handle_rendering_data` synthesizes images for training pose estimation.
